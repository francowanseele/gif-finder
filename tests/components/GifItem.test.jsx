import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test <GifItem />", () => {
  const title = "Homero Simpson";
  const url = "https://thesimpsons/homero-simpson.jpg";

  test("Should match with snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Should show the image with url and ALT", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Should show title in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
