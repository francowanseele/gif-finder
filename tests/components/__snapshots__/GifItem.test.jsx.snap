// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifItem /> Should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Homero Simpson"
      src="https://thesimpsons/homero-simpson.jpg"
    />
    <p>
      Homero Simpson
    </p>
  </div>
</div>
`;
